generator/src/test/java/com/scottlogic/deg/generator/cucumber/After.feature

```diff
@@ -0,0 +1,323 @@
+    And the generator can generate at most 4 rows
+  Then the following data should be generated:
+    | foo                     |
+    | 2018-09-01T00:00:00.001 |
+    | 2018-09-01T00:00:00.002 |
+    | 2018-09-01T00:00:00.003 |
+    | 2018-09-01T00:00:00.004 |
+
+Scenario Outline: 'After' invalid datetime fails with error
+  Given foo is after <dateValue>
+  Then I am presented with an error message
+    And no data is created
+  Examples:
+    | dateValue               |
+    | 2018-09-32T00:00:00.000 |
+    | 2018-09-01T25:00:00.000 |
+    | 2018-13-01T00:00:00.000 |
+
+#Issue #610 - Leap year date invalid
+@ignore
+Scenario: 'After' non-existent leap year date fails with error
+  Given foo is after 2019-02-29T00:00:00.000
+  Then I am presented with an error message
+  And no data is created
+
+  ### after ###
+
+# Defect 611 "before run against a non contradicting before results in an error" related to this scenario
+@ignore
+Scenario: 'After' with a non-contradicting 'After' is successful
+  Given foo is after 2019-01-01T00:00:00.000
+    And foo is after 2019-02-01T00:00:00.000
+    And the generator can generate at most 5 rows
+  Then the following data should be generated:
+    | foo                     |
+    | 2019-02-01T00:00:00.001 |
+    | 2019-02-01T00:00:00.002 |
+    | 2019-02-01T00:00:00.003 |
+    | 2019-02-01T00:00:00.004 |
+    | 2019-02-01T00:00:00.005 |
+
+#Related to #667 - Expected time increment is one millisecond
+@ignore
+Scenario: 'After' with a non-contradicting 'Not After' is successful
+  Given foo is after 2019-01-01T00:00:00.000
+    And foo is anything but after 2020-01-01T00:00:00.000
+    And the generator can generate at most 5 rows
+  Then the following data should be generated:
+    | foo                     |
+    | 2019-01-01T00:00:00.001 |
+    | 2019-01-01T00:00:00.002 |
+    | 2019-01-01T00:00:00.003 |
+    | 2019-01-01T00:00:00.004 |
+    | 2019-02-01T00:00:00.005 |
+
+# Defect 611 "before run against a non contradicting before results in an error" related to this scenario
+@ignore
+Scenario: 'Not After' with a non-contradicting 'Not After' is successful
+  Given foo is anything but after 2019-01-01T00:00:00.000
+    And foo is anything but after 2020-01-01T00:00:00.000
+    And the generator can generate at most 5 rows
+  Then the following data should be generated:
+    | foo                     |
+    | 2019-01-01T00:00:00.000 |
+    | 2018-12-31T23:59:59.999 |
+    | 2018-12-31T23:59:59.998 |
+    | 2018-12-31T23:59:59.997 |
+    | 2018-12-31T23:59:59.996 |
+
+
+### Contradictions ###
+
+#Defect #672 - Request for contradictory After and Not After should only generate null
+@ignore
+Scenario: 'After' with a contradicting 'Not After' generates null
+  Given foo is after 2019-01-01T00:00:00.000
+    And foo is anything but after 2019-01-01T00:00:00.000
+  Then the following data should be generated:
+    | foo                     |
+    | null                    |
+
+
```

Comment:
Try to be consistent with whitespace - I think we just want one blank line between tests

Reply:
Edited file for whitespace consistency